sdk/search/azure-search/src/main/java/com/azure/search/DataSources.java

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.search.models.DataChangeDetectionPolicy;
+import com.azure.search.models.DataContainer;
+import com.azure.search.models.DataDeletionDetectionPolicy;
+import com.azure.search.models.DataSource;
+import com.azure.search.models.DataSourceCredentials;
+import com.azure.search.models.DataSourceType;
+import com.azure.search.models.HighWaterMarkChangeDetectionPolicy;
+
+/**
+ * Utility class generating DataSource object per DataSourceType
+ */
+public class DataSources {
+
+    /**
+     * Creates a new DataSource to connect to an Azure SQL database.
+     *
+     * @param name The name of the data source.
+     * @param sqlConnectionString The connection string for the Azure SQL database.
+     * @param tableOrViewName The name of the table or view from which to read rows.
+     * @param description Optional. Description of the data source.
+     * @param changeDetectionPolicy The change detection policy for the data source.
+     * Note that only high watermark change detection
+     * is allowed for Azure SQL when deletion detection is enabled.
+     * @param deletionDetectionPolicy Optional. The data deletion detection policy for the data source.
+     * @return A new DataSource instance.
+     */
+    public static DataSource azureSql(
+        String name,
+        String sqlConnectionString,
+        String tableOrViewName,
+        String description,
+        DataChangeDetectionPolicy changeDetectionPolicy,
+        DataDeletionDetectionPolicy deletionDetectionPolicy) {
+        return new DataSource()
```

Comment:
nit: I feel like the parameter variables should be in the same order as you add them to the object

Reply:
modifed